### ISSUE TITLE
GM.notification returns promise that never get fulfilled

### ISSUE THREAD

**Issue by 7nik** [Tampermonkey/tampermonkey#999]
GM.notification returns a promise but it doesn't get resolved or rejected even after clicking/closing the notification.

It would nice to resolve the promise on `done` event.

## Expected Behavior
The userscript below logs smth to the console.

## Actual Behavior
The is nothing in the console.

## Specifications

- Chromium: Version 83.0.4103.97 (Official Build) Arch Linux (64-bit)
- TM: stable, 4.10
- OS: Manjaro Linux (64-bit)

## Script

```js
// ==UserScript==
// @name         Test notification
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @grant        GM.notification
// ==/UserScript==

(async () => {
    try {
        console.log("returns:", await GM.notification({ text: "test message" }));
    } catch (ex) {
        console.error("error:", ex);
    }
})();
```

**Comment by 7nik** [comment]
Also,
1. The docs don't say that  `GM.notification(text, title, image, onclick)` is not supported.
2. The docs don't say that `ondone` receives boolean argument `clicked`.
3. No defaulting of `title` to `@name` and `image` to `@icon`/`@icon64` like in the GM_notification.

**Comment by derjanb** [comment]
Fixed at TM BETA 4.11.6116

**Comment by 7nik** [comment]
👍
